### PR TITLE
Unit tests for IQ

### PR DESCRIPTION
**Pull request by DarthHater**
Welp, looks like we missed unit tests for IQ way back, so here is the start

This pull request makes the following changes:
* Adds `test_iq.py`
* Adds a test for getting the `internal_application_id` from IQ, passes
* Adds a test for getting a response from `submit_sbom`, very rudimentary, passes
* Adds a test for getting a response from `poll_report`, a `None` case and a `Failure` case. One fails, this is by design to help us figure out another issue @bhamail was looking at
* Adds `responses` library`, way easier to use than `unittest.mock`, especially for something that calls multiple URLs

As well, this fixes the issue that @bhamail was saying, by being more explicit about Policy Actions, and covering this in `__main__.py`

cc @bhamail / @DarthHater

### CODE REVIEW

File: jake/test/test_iq.py [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+    """test_call_get_application_id mocks a call to IQ
+    and ensures that that calls to IQ for an application ID return
+    an internal ID as expected"""
+    file = Path(__file__).parent / "iqapplicationresponse.txt"
+    with open(file, "r") as stdin:
+      responses.add(responses.GET,
+                    'http://afakeurlthatdoesnotexist.com:8081/api/v2/applications?publicId=testapp',
+                    body=stdin.read(), status=200)
+      response = self.func.get_internal_id()
+    self.assertEqual(len(response), 32)
```
Comment (bhamail) [inline jake/test/test_iq.py:72]: super minor: Curious why the `len(response)` check before the literal string check.

File: jake/iq/iq.py [hunk 2]
```diff
@@ -136,6 +137,7 @@ def __handle_response(self, response: str) -> (bool):
       res = json.loads(response)
       LOG.debug(res)
       if res['policyAction'] == 'None':
+        self._policy_action = res['policyAction']
```
Comment (bhamail) [inline jake/iq/iq.py:140]: minor: seems like `self._policy_action = res['policyAction']` always happens regardless of the `res['policyAction']` value, so maybe it should be set outside the `if / else`. e.g.

```
      self._policy_action = res['policyAction']
      if res['policyAction'] == 'None':
        LOG.debug("No policy issues, whew!")
      self._report_url = res['reportHtmlUrl']
```

**Review by bhamail — APPROVED** [review]

File: jake/iq/iq.py (hunk 2 above)
Comment (DarthHater) [reply jake/iq/iq.py]: How this? 3c1a35c

File: jake/test/test_iq.py (hunk 1 above)
Comment (DarthHater) [reply jake/test/test_iq.py:72]: I got these tests basic structure from the `test_ossindex.py` file. I can drop this!

File: jake/iq/iq.py (hunk 2 above)
Comment (bhamail) [reply jake/iq/iq.py]: PERFECTO!

Comment (bhamail) [reply jake/iq/iq.py]: CAPS LOCK FRIDAY IN PR - FOREVER

File: jake/test/test_iq.py (hunk 1 above)
Comment (DarthHater) [reply jake/test/test_iq.py]: In: 7a3772f

Comment (bhamail) [reply jake/test/test_iq.py]: NOICE! EVEN BETTER

**Pull request merged**